test/stdlib/NewString.swift

```diff
@@ -14,45 +14,33 @@ func hexAddrVal<T>(_ x: T) -> String {
   return "@0x" + hex(UInt64(unsafeBitCast(x, to: UInt.self)))
 }
 
-func hexAddr(_ x: AnyObject?) -> String {
-  if let owner = x {
-    if let y = owner as? _HeapBufferStorage<_StringBufferIVars, UInt16> {
-      return ".native\(hexAddrVal(y))"
-    }
-    if let y = owner as? NSString {
-      return ".cocoa\(hexAddrVal(y))"
-    }
-    else {
-      return "?Uknown?\(hexAddrVal(owner))"
-    }
-  }
-  return "null"
-}
-
 func repr(_ x: NSString) -> String {
   return "\(NSStringFromClass(object_getClass(x)))\(hexAddrVal(x)) = \"\(x)\""
 }
 
-func repr(_ x: _StringCore) -> String {
-  if x.hasContiguousStorage {
-    if let b = x.nativeBuffer {
-    var offset = x.elementWidth == 2
-      ? b.start - UnsafeMutableRawPointer(x.startUTF16)
-      : b.start - UnsafeMutableRawPointer(x.startASCII)
-      return "Contiguous(owner: "
-      + "\(hexAddr(x._owner))[\(offset)...\(x.count + offset)]"
-      + ", capacity = \(b.capacity))"
-    }
-    return "Contiguous(owner: \(hexAddr(x._owner)), count: \(x.count))"
-  }
-  else if let b2 = x.cocoaBuffer {
-    return "Opaque(buffer: \(hexAddr(b2))[0...\(x.count)])"
+func repr(_ x: _StringGuts) -> String {
+  if x._isNative {
+    return "Native("
+      + "owner: \(hexAddrVal(x._owner)), "
+      + "count: \(x.count), "
+      + "capacity: \(x.capacity))"
+  } else if x._isNonTaggedCocoa {
+    return "Cocoa("
+      + "owner: \(hexAddrVal(x._owner)), "
+      + "count: \(x.count))"
+  } else if x._isTaggedCocoa {
```

Comment:
@lorentey I believe the structure changed around here (i.e. there is no more _isTaggedCocoa var).